tsl/src/nodes/vector_agg/grouping_policy_hash.c

```diff
@@ -317,24 +309,24 @@ gp_hash_add_batch(GroupingPolicy *gp, DecompressBatchState *batch_state)
 	for (int i = 0; i < policy->num_grouping_columns; i++)
 	{
 		const GroupingColumn *def = &policy->grouping_columns[i];
-		const CompressedColumnValues *values = &batch_state->compressed_columns[def->input_offset];
-		policy->current_batch_grouping_column_values[i] = *values;
-	}
 
+		policy->current_batch_grouping_column_values[i] =
+			vector_slot_get_compressed_column_values(vector_slot,
+													 AttrOffsetGetAttrNumber(def->input_offset));
```

Comment:
To illustrate the thing we're discussing elsewhere, this is the absolutely crucial point that you have to keep to avoid conflicts. And if you can do that, you probably can trivially do that in all other places that use the `CompressedColumnValues`  and make less changes?

Reply:
We need to standardize on the data representation and produce it from all the slot types, ideally in the fewest number of places possible, so that all the rest of the code doesn't have to dispatch on the slot type.

Reply:
The vector_slot functions was intended to be the common interface because arrow slots don't need CompressedColumnValues in most cases. I'll go with CompressedColumnValues for now since I don't want to conflict with your other work, but I really do think that it represents an extra layer of indirection which is not needed.